precommit/src/main/shell/personality/hadoop.sh

```diff
@@ -404,11 +404,34 @@ function personality_modules
   fi
 
   extra="-Ptest-patch ${extra}"
-
-  for module in $(hadoop_order ${ordering}); do
-    # shellcheck disable=SC2086
-    personality_enqueue_module ${module} ${extra}
+  OZONE_CHANGED=false
+  CORE_HADOOP_CHANGED=false
+  for module in $CHANGED_MODULES
+  do
+    if [[ "$module" =~ "hdds" ]]; then
+      OZONE_CHANGED=true
+    elif [[ "$module" =~ "ozone" ]]; then
+      OZONE_CHANGED=true
+    else
+      CORE_HADOOP_CHANGED=true
+    fi
   done
+
+  if [ "$OZONE_CHANGED" = true ]; then
+    extra="-Phdds ${extra}"
+  fi
+
+  if [ "$CORE_HADOOP_CHANGED" = false ] && [ "$OZONE_CHANGED" = true ]; then
+    if [ "$testtype" != "mvnsite" ] && [ "$testtype" != "shadedclient" ]; then
+      personality_enqueue_module hadoop-hdds ${extra}
```

Comment:
shellcheck disable the message here.  extra should have been an array but it wasn't when we first wrote yetus.